sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewViewModel.groovy

```diff
@@ -65,20 +75,25 @@ class ProjectOverviewViewModel {
         projectOverviews.remove(projectOverview)
     }
 
-    private void updateSamplesReceived(String projectCode, int sampleCount) {
-        getProjectSummary(projectCode).samplesReceived = sampleCount
+    private void updateTotalSampleCount(StatusCount statusCount){
+        ProjectSummary summary = getProjectSummary(statusCount.projectCode)
+        summary.totalSampleCount = statusCount.totalSampleCount
     }
 
-    private void updateDataAvailable(String projectCode, int sampleCount) {
-        getProjectSummary(projectCode).sampleDataAvailable = sampleCount
+    private void updateSamplesReceived(StatusCount statusCount) {
+        getProjectSummary(statusCount.projectCode).samplesReceived = statusCount.count
```

Comment:
Because I wanted to reduce the input params. Otherwise I would have been necessary to add a third input param for each of the methods. I kept it as it was but I can introduce the variable, I like that!